pkg/mdformatter/linktransformer/link.go

```diff
@@ -150,15 +150,20 @@ func NewValidator(logger log.Logger, except *regexp.Regexp, anchorDir string) (m
 	}); err != nil {
 		return nil, err
 	}
+	v.c.OnRequest(func(request *colly.Request) {
+		v.rMu.Lock()
+		defer v.rMu.Unlock()
+		request.Ctx.Put("originalURL", request.URL.String())
```

Comment:
Nice!

Suggestion: Handcrafted strings like this can be prone to errors. Use constant to reuse such key. Also it does not need to be string to be key. It's recommended to use empty struct (less mem allocated) e.g

`var originalURLKey = struct{}{} `

Reply:
Sure!

Reply:
The `Get` and `Put` methods seem to only take in strings as highlighted [here](https://github.com/gocolly/colly/blob/master/context.go#L23). Should `originalURLKey` be passed in using `fmt.Sprintf` here? But that maybe wouldn’t help with memory as that would be the same as using a string…

Reply:
Ah right, this is something library specific not Go context.. ok, let's do constant at least